packages/monaco/src/browser/monaco-quick-input-service.ts

```diff
@@ -0,0 +1,173 @@
+    password?: boolean;
+
+    /**
+     * Set to `true` to keep the input box open when focus moves to another part of the editor or to another window.
+     */
+    ignoreFocusOut?: boolean;
+
+    /**
+     * An optional function that will be called to validate input and to give a hint
+     * to the user.
+     *
+     * @param value The current value of the input box.
+     * @return Return `undefined`, or the empty string when 'value' is valid.
+     */
+    validateInput?(value: string): string | undefined | PromiseLike<string | undefined>;
+}
+
+const promptMessage = "Press 'Enter' to confirm your input or 'Escape' to cancel";
+
+@injectable()
+export class QuickInputService {
+
+    private opts: MonacoQuickInputControllerOptsImpl;
+    @inject(MonacoQuickOpenService)
+    protected readonly quickOpenService: MonacoQuickOpenService;
+
+    open(options: QuickInputOptions): Promise<string | undefined> {
+        options.prompt = this.createPrompt(options.prompt);
+
+        const inputItem = new InputOpenItemOptions(options.prompt);
+        this.opts = new MonacoQuickInputControllerOptsImpl({
+            onType: (s, a) => this.validateInput(s, a, inputItem)
+        },
+            options,
+            {
+                prefix: options.value,
+                placeholder: options.placeHolder,
+                onClose: () => inputItem.resolve(undefined)
+            });
+        this.quickOpenService.internalOpen(this.opts);
+
+        return new Promise(r => {
+            inputItem.resolve = r;
+        });
+    }
+
+    private createPrompt(prompt?: string): string {
+        if (prompt) {
+            return `${prompt} (${promptMessage})`;
+        } else {
+            return promptMessage;
+        }
+    }
+
+    private validateInput(str: string, acceptor: (items: QuickOpenItem[]) => void, inputItem: InputOpenItemOptions): void {
+        inputItem.currentText = str;
+        acceptor([new QuickOpenItem(inputItem)]);
+        if (this.opts && this.opts.validateInput) {
+            const hint = this.opts.validateInput(str);
+            if (hint) {
+                if (typeof hint !== 'string') {
+                    hint.then(p => {
+                        if (p) {
+                            this.setValidationState(inputItem, p, false);
+                        } else {
+                            this.setValidationState(inputItem, this.opts!.prompt!, true);
+                        }
+                    });
+                } else {
+                    this.setValidationState(inputItem, hint, false);
+                }
+            } else {
+                this.setValidationState(inputItem, this.opts.prompt!, true);
+            }
+        }
+    }
+
+    private setValidationState(inputItem: InputOpenItemOptions, label: string, isValid: boolean): void {
+        this.quickOpenService.clearInputDecoration();
+        inputItem.isValid = isValid;
+        inputItem.label = label;
+        // this.widget.refresh();
```

Comment:
should we remove this line ?

Reply:
Yes, sorry I merge it before saw your comment :(

Reply:
Just in case you fix this, could you please update the documentation too?

https://github.com/theia-ide/theia/pull/2161/files#diff-c12f4cfff29006870b5f6654b73c2f8dR1511

Reply:
I will update https://github.com/theia-ide/theia/pull/2185 with removing commented line